cmd/kubeops/internal/handler/dashboardcompat.go

```diff
@@ -0,0 +1,121 @@
+// The Dashboard doesn't understand the Helm 3 release format.
+// This file is a compatibility layer that translates Helm 3 releases to a Helm 2-similar format suitable for the Dashboard.
+
+package handler
+
+import (
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/any"
+	"gopkg.in/yaml.v2"
+	h3chart "helm.sh/helm/v3/pkg/chart"
+	h3 "helm.sh/helm/v3/pkg/release"
+	h2chart "k8s.io/helm/pkg/proto/hapi/chart"
+	h2 "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+// generatedYamlHeader is prepended to YAML generated from the internal map[string]interface{} representation.
+const generatedYamlHeader = "# Not original YAML! Generated from parsed representation."
+
+type dashboardCompatibleRelease struct {
```

Comment:
Is there an issue using the `h2.Release` instead of creating a new struct here? It appears to have the same fields (extra ones serialize with `omitempty`):

https://github.com/kubeapps/kubeapps/blob/master/pkg/agent/agent_test.go#L185

Reply:
We originally wanted to write a function from `h3.Release` to `h2.Release`, but arrived at the conclusion that it's impossible in general. Can't remember exactly why, but I think there were several reasons. One might be that the respective `Metadata` types are not isomorphic. The `Values` property in the respective `Chart` types also seems problematic.

I've, however, made a best-effort attempt at using Helm 2 types instead of custom ones. It seems to work; see d67e136.

Reply:
Right, here we're just looking at getting data back to the dashboard for presentation, rather than interacting with helm, so I don't *think* isomorphism is essential as it would be if we were generating this for use with helm itself. Thanks for the update!